### PR TITLE
Include newsletter

### PR DESCRIPTION
**Pull request by hngerebara**
Please view behavior [here](https://6dd3ffcc.ngrok.io/guides/) 
Try subscribing as well.

https://drive.google.com/file/d/1svUYMclM88880paBVAynbyTRGple6-4W/view?usp=sharing

PS. The URL for displaying the confirmation page will be changed to our website after it's been merged. this is only for testing.

### CODE REVIEW

**Comment by brikis98** [comment]
Is the URL after subscribing something we can configure dynamically or only statically in MailChimp? If static, send users to the home page with some thank you dialog; if dynamic, send users back to whatever page they were on when subscribing.

**Comment by hngerebara** [comment]
We can configure dynamically in MailChimp so I'll do the latter

**Comment by hngerebara** [comment]
This is now ready for review
https://drive.google.com/file/d/1svUYMclM88880paBVAynbyTRGple6-4W/view?usp=sharing

**Comment by Etiene** [comment]
Just a small nitpick, the subscribe button can be one more pixel higher

File: _includes/newsletter-modal.html [hunk 1]
```diff
@@ -4,8 +4,8 @@
             <div class="modal-body">
                 <img src="/assets/img/guides/layout/img-newsletter.png" alt="newsletter_image" /><br />
                 <label class="modal-title" id="newsletterModalLabel">Success! You’re signed up. </label>
-                <p>You signed up for our newsletter with the email address: </p>
-                <button type="button" class="btn btn-secondary" data-dismiss="modal">Got it</button>
+                <p>You signed up for our newsletter </p>
+                <button type="button" class="btn btn-secondary" onclick="window.history.pushState({path: window.location.pathname },'',window.location.pathname)" data-dismiss="modal">Got it</button>
```
Comment (brikis98) [inline _includes/newsletter-modal.html]: Why do you need this `onclick` behavior here?

File: _includes/newsletter.html [hunk 2]
```diff
@@ -1,24 +1,15 @@
...
-                <div class="input-group">
-                    <input type="text" id="email" class="form-control guides-input-box" placeholder="Your Email Address">
-                    <div class="input-group-btn">
-                        <div class="btn-group" role="group"> 
-                            <button type="button" id="newsletter-subscribe" class="btn btn-primary" ga-event-category="newsletter" ga-event-action="nl-subscribe">
-                                Subscribe
-                            </button>
-                        </div>
-                    </div>
+            <form  id="guides-newsletter-form"class="col-xs-10 col-xs-offset-1" method="post" action="//gruntwork.us13.list-manage.com/subscribe/post?u=8ccaad0747ac13944cf318e0f&amp;id=c19e11ead2" target="_blank">
```
Comment (brikis98) [inline _includes/newsletter.html]: I noticed the `id=...` param changed. Is that intentional?

File: _includes/search.html [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<div class="row">
+<div class="row" id="filter-section">
```
Comment (brikis98) [inline _includes/search.html]: Is this change related to this PR?

File: assets/js/guides.js [hunk 4]
```diff
@@ -1,4 +1,9 @@
 $(document).ready(function () {
+
+  if (window.location.href.indexOf('newsletter_success') !== -1) {
+    $("#newsletter-success").modal('show');
+  }
```
Comment (brikis98) [inline assets/js/guides.js]: This check should be in `main.js` so it affects all pages, not just guides.

File: pages/guides/_no-search-results.html [hunk 5]
```diff
@@ -1,12 +1,8 @@
...
-    <div class="row text-center">
-        <div class="no-results no-results-space">
-            <div class="col-md-6 col-md-offset-3" >
-                <img src="/assets/img/guides/layout/no-search-results.png" alt="no search results" /><br />
-                <label>No Results Found</label>
-                <p>We couldn’t find any guides matching your search. Contact us and we can help you find the answers you’re looking for.</p>
-                <a href="/contact"><button type="button" class="btn btn-primary">Contact Us</button></a>
-            </div>
-        </div>
+<div id="no-matches" class="no-results no-results-space center">
```
Comment (brikis98) [inline pages/guides/_no-search-results.html]: This change doesn't seem related to this PR

Comment (hngerebara) [reply pages/guides/_no-search-results.html]: No it's not. I noticed it wasn't well-positioned in the middle while I was working on this and decided to fix it at once. I can take it out and create a separate PR.

File: assets/js/guides.js (hunk 4 above)
Comment (hngerebara) [reply assets/js/guides.js]: Okay

File: _includes/newsletter-modal.html (hunk 1 above)
Comment (hngerebara) [reply _includes/newsletter-modal.html]: I added it to clear the URL to point to the current path once the button is clicked
E.g When we have the additional parameters in the URL `newsletter_success=1`, after clicking on the `Got it` button which dismisses the modal, the URL remains the same having the url like `/guides?newsletter_success=1. So I did that to ensure that only the pathname is displayed without the parameters.

File: _includes/newsletter.html (hunk 2 above)
Comment (hngerebara) [reply _includes/newsletter.html]: Yes, it was. I wanted to specify that this is coming from the guides page. In case we want to make reference to it anytime.

File: _includes/search.html (hunk 3 above)
Comment (hngerebara) [reply _includes/search.html]: I included it when I was trying to hide sections before I clarified during the standup. I'll take it out now

**Comment by hngerebara** [comment]
Ooh crap. I was subscribing them to the `Gruntwork Subscribers Newsletter`. I'll fix that now.

Yes, I fixed the styling issue in this PR. I previously added a `.newsletter` class and styled it mainly for guides page only to realize yesterday that it was the same class name used for styling the /newsletter page. I fixed that already here[](https://github.com/gruntwork-io/gruntwork-io.github.io/pull/177/files#diff-1e69babe6c50a615d3a6ef7b48cf7484R1366). Once this is merged, it will fix that.

File: _includes/newsletter-modal.html (hunk 1 above)
Comment (brikis98) [reply _includes/newsletter-modal.html]: I see. Can you add that explanation as a comment?

**Comment by hngerebara** [comment]
Updated this 
https://drive.google.com/file/d/1jj60uhN5fLmCZxl6DXGuwtZ5XqDr83O2/view?usp=sharing
